### DISCUSSION TITLE
New UI (Where to put this?)

### DISCUSSION THREAD

**Discussion by udippel** [gerbera/gerbera#3072] [General]
Under Pull requests wouldn't be the correct location. 
Whatever. Looks splendidly more modern and logical. Well done!

Though, I don't think the side-by-side of two boxes with different heights would be the last word on UI. 

How about the 'modern' three horizontal lines for 'menu'; thinking that in 2024 everyone understands this, somewhere in the "Welcome to gerbera ..." area? and then not duplicate the 3 menu items, horizontally in the top line and large and iconized in the left box, instead leaving (for the configured case) just the Server Statistics on the welcome page?

Maybe it's only here, though my welcome page looks like a strange mixture of fonts, sizes, colours; with the version number unreadable except by marking? I think that all this (version, contributors, licences, ...) could better fit under that menu, e.g. "About".
How does everyone else think about this?

![image](https://github.com/gerbera/gerbera/assets/50265752/06bb75a0-118e-4b91-9e43-acb5b1eb322e)

**Comment by whyman** [comment]
Patches are welcome, or diagrams, because I am not sure words are the right media for such things

**Reply by udippel** [reply]
Sorry, I am not an artist nor UI designer. 
I tried a very rough diagram on what I meant as starting point, though.
The additional 'page' needed would be the 'About', for the rest it would work like the current menu(s). 
Preferably the menu slides into 'off', ie right margin, on click. I couldn't make up my mind if the three lines should be in the upper field, like in the drawing, or slightly below in the field with the gerbera background. 

In case this thought is pursued, one might as a second step do away with the narrow upper field altogether and display server name, gerbera (there is one by now, very large, background) on the field with the gerbera background, and a menu handle. 
To me, this would look even more modern. 
![Gerb_intermediate_1](https://github.com/gerbera/gerbera/assets/50265752/6a7c1d94-5386-4a9f-be16-1afc25729dc3)

**Reply by cweiske** [reply]
I do not like to move links into a hidden menu when there is enough screen estate for them. It's one click more to use them.

Also having them duplicated beside the server statistics is fine, because new users get an explanation what they can do. Knowledgable users can simply use the main menu on top.

**Reply by udippel** [reply]
I understand. 
From a HCI point of view, the upper window border is totally cluttered, size wise unreadable, at least here. Different fonts and colours, even super- and subscript-look of machine name.
For me it is the opposite, though: my interest in the number of files and data is quite low. At my end, I have to click-click-click to see what I am actually interested in: Recent modifications, additions, etc. Therefore I added some items to the existing four as examples on the suggested right hand pull-down, comparably to what a 'normal' Settings usually offers. 

Maybe it is only me, but here the web interface is the most used interface, except of where BubbleUPnP exists. Kodi is somewhat clunky, like for babies, giant icons, not very stable. On W10 it is VLC for us, and there access is rather cumbersome through Playlists.
On the web interface, the tree structure is clearly visible, and the majority of files just plays at clicking on the database entry. Not perfect, but quite good. Had it a "Search" included, I'd throw away anything else. 

Since we are at it, one might as well ponder of having an open part of the gerbera web interface, for access of data (read only); combined with a login to show the administrative interface as we see it in front of us now. Like showing the Database in any case, and everything else after login only. 

HCI-wise: Inconsistent naming issues ought to be remedied: "Database" vs. "Browse indexed Files"; similar to "Files System" and "Add Files". Also therefore I'd prefer a single point of entry. Also for the new user.

**Reply by udippel** [reply]
"... into a hidden menu ..." I agree with you absolutely, and stand corrected w.r.t. the menu descriptor. For the rest, so far I stick to my opinion.
